soh/src/overlays/misc/ovl_kaleido_scope/z_kaleido_collect.c

```diff
@@ -564,21 +627,55 @@ void KaleidoScope_DrawQuestStatus(PlayState* play, GraphicsContext* gfxCtx) {
 
                 if (pauseCtx->unk_1E4 == 8) {
                     s16 Notes_alpha = 200;
-                    if (gOcarinaSongNotes[sp224].notesIdx[phi_s3] == 0) {
-                        gDPSetPrimColor(POLY_KAL_DISP++, 0, 0, aButtonColor.r, aButtonColor.g, aButtonColor.b, Notes_alpha);
+                    // Quick hack for getting custom colors to work with custom ocarina controls
+                    switch (sOcarinaBtnMaps[gOcarinaSongNotes[sp224].notesIdx[phi_s3]]) {
+                        case BTN_A:
+                            gDPSetPrimColor(POLY_KAL_DISP++, 0, 0, aButtonColor.r, aButtonColor.g, aButtonColor.b,
+                                            Notes_alpha);
+                            break;
+                        case BTN_CDOWN:
+                            gDPSetPrimColor(POLY_KAL_DISP++, 0, 0, cDownButtonColor.r, cDownButtonColor.g,
+                                            cDownButtonColor.b, Notes_alpha);
+                            break;
+                        case BTN_CRIGHT:
+                            gDPSetPrimColor(POLY_KAL_DISP++, 0, 0, cRightButtonColor.r, cRightButtonColor.g,
+                                            cRightButtonColor.b, Notes_alpha);
+                            break;
+                        case BTN_CLEFT:
+                            gDPSetPrimColor(POLY_KAL_DISP++, 0, 0, cLeftButtonColor.r, cLeftButtonColor.g,
+                                            cLeftButtonColor.b, Notes_alpha);
+                            break;
+                        case BTN_CUP:
+                            gDPSetPrimColor(POLY_KAL_DISP++, 0, 0, cUpButtonColor.r, cUpButtonColor.g, cUpButtonColor.b,
+                                            Notes_alpha);
+                            break;
+                        default:
+                            gDPSetPrimColor(POLY_KAL_DISP++, 0, 0, cButtonsColor.r, cButtonsColor.g, cButtonsColor.b,
+                                            Notes_alpha);
+                            break;
+                    }
+                    /// OLD FUNCTIONALITY
```

Comment:
```suggestion
                    /// OLD FUNCTIONALITY
                    // todo: remove if not needed
```

just adding this to kick off a build